sympy/sets/tests/test_sets.py

```diff
@@ -14,17 +14,30 @@
 
 def test_imageset():
     ints = S.Integers
+    assert imageset(x, x - 1, S.Naturals) is S.Naturals0
+    assert imageset(x, x + 1, S.Naturals0) is S.Naturals
+    assert imageset(x, abs(x), S.Naturals0) is S.Naturals0
+    assert imageset(x, abs(x), S.Naturals) is S.Naturals
+    assert imageset(x, abs(x), S.Integers) is S.Naturals0
+    # issue 16878a
+    r = symbols('r', real=True)
+    assert (1, r) in imageset(x, (x, x), S.Reals)
```

Comment:
Maybe I misunderstand: this is the set `(x,x)` so `(1,r)` is only in the set if `r=1` and we don't know that r is 1...

Reply:
hmm...agree